### PR TITLE
render/animation: Add `animation.Animate` widget.

### PR DESCRIPTION
**Pull request by fxb**
### Description

The `animation.Animate` widget animates a child widget by interpolating between transforms (translate, scale, rotate) using keyframes similar to how CSS transforms and animations work.

This also add  `Curve` parsing:
  * Implement `ParseCurve`, which can handle predefined and custom cubic bezier curves.
  * Add `CustomCurve` which accepts a Starlark provided function.

### CODE REVIEW

**Comment by fxb** [comment]
@matslina

File: render/animation/sequence.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package animation
+
+import (
+	"image"
+
+	"github.com/fogleman/gg"
+	"tidbyt.dev/pixlet/render"
+)
+
+// Sequence renders a list of `Animate` widgets in sequence.
```
Comment (fxb) [inline render/animation/sequence.go:10]: Actually, I believe this widget could live as `render.Sequence` and be a more advanced version of `render.Animation`, supporting animated children.

File: render/marquee.go [hunk 2]
```diff
@@ -73,7 +74,7 @@ func (m Marquee) FrameCount() int {
 		offend = -cw
 	}
 
-	return cw + offstart + size - offend + 1
+	return cw + offstart + size - offend
```
Comment (fxb) [inline render/marquee.go:77]: Fixed the extra "jump" in `render.Marquee` that some people were complaining about.

File: docs/animation.md [hunk 3]
```diff
@@ -5,10 +5,112 @@
...
+
+When translating the child widget on the X- or Y-axis, it often is
+desireable to round to even integers, which can be controlled via
+`rounding`, which defaults to `round`. Possible values are `round` to
+round to the nearest integer, `floor` to round down, `ceil` to round
+up or `none` to not perform any rounding. Rounding only is applied for
+translation transforms, but not to scaling or rotation transforms.
+
+If `wait_for_child` is set to `True`, the animation will finish and
+then wait for all chuld frames to play before restarting. If it is set
```
Comment (matslina) [inline docs/animation.md:66]:
```suggestion
then wait for all child frames to play before restarting. If it is set
```

**Pull request merged**